### PR TITLE
Community typings for Polymer V2

### PR DESCRIPTION
**Pull request by Buslowicz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by Draccoz** [comment]
Could anybody please hint me on how to fix that PR? I do not have tests file, I'd rather not write those (at least for now).

**Comment by andy-ms** [comment]
`npm info alter-polymer` turns up nothing -- what specific package is this for? Use a more specific link than `polymer-project.org`, which is about the project as a whole.

File: types/alter-polymer/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "es2015",
+    "module": "commonjs",
+     "lib": [
+       "dom",
+       "es2015"
+     ],
+    "strict": true,
+    "noImplicitAny": false,
```
Comment (andy-ms) [inline types/alter-polymer/tsconfig.json]: These should be `true` for new projects.

File: types/alter-polymer/tslint.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "variable-name": false,
```
Comment (andy-ms) [inline types/alter-polymer/tslint.json]: New projects should have a clean lint.

File: types/alter-polymer/tsconfig.json (hunk 1 above)
Comment (Draccoz) [inline types/alter-polymer/tsconfig.json]: Ok, I'll make it all strict.

File: types/alter-polymer/tslint.json (hunk 2 above)
Comment (Draccoz) [inline types/alter-polymer/tslint.json]: What if my types need properties/methods starting with underscore?

**Comment by Draccoz** [comment]
Thanks for reply.
These are "community driven" types for Polymer V2. Official types will be probably provided within the package itself (or not), but are not ready yet. "alter-polymer" stands for "Alternative Polymer", I want to avoid taking anything closer to original name as I don't want to block official types later.
As for `@types/polymer`, these were types for Polymer V1 and are not compatible with V2.
Should I point a `Project` link to github repo?

**Comment by andy-ms** [comment]
We support having multiple major versions of a package side-by-side, so you can just move the present `polymer` definitions into `types/polymer/v1` and put the new types in `types/polymer`. However, we would remove it once polymer has its own types.

>  Official types will be probably provided within the package itself (or not), but are not ready yet.

Why not just contribute there?

File: types/alter-polymer/tslint.json (hunk 2 above)
Comment (andy-ms) [inline types/alter-polymer/tslint.json]: Use `// tslint:disable-next-line variable-name` as necessary.

**Comment by Draccoz** [comment]
Releasing it separately will work for most cases and will be finished now. If I were to contribute to original repo, it would take longer.
If having these types under this name is a problem, I can release them under `twc` (which is my project, which depends on those types), though I would like to keep those separate as well.

**Comment by andy-ms** [comment]
I don't want to compete with the official definitions, even if they are slow. You could just update `polymer` itself, and we'll deprecate that once they publish their own typings.

**Comment by Draccoz** [comment]
Again, I will not interfere with official typings. Polymer team wants to have types generated and in-the-package, and I wont stop them. If releasing that under different name as temporary types (which people ask for) is not possible, I will release those as a regular npm package.

**Pull request closed without merging**